src/measure_theory/integral/set_to_l1.lean

```diff
@@ -0,0 +1,605 @@
+define the Bochner integral in the `bochner` file. It will also be used to define the conditional
+expectation of an integrable function (TODO).
+
+## Main Definitions
+
+- `fin_meas_additive μ T`: the property that `T` is additive on measurable sets with finite measure.
+  For two such sets, `s ∩ t = ∅ → T (s ∪ t) = T s + T t`.
+- `dominated_fin_meas_additive μ T C`: `fin_meas_additive μ T ∧ ∀ s, ∥T s∥ ≤ C * (μ s).to_real`.
+  This is the property needed to perform the extension from indicators to L1.
+- `set_to_L1 (hT : dominated_fin_meas_additive μ T C) : (α →₁[μ] E) →L[ℝ] F`: the extension of `T`
+  from indicators to L1.
+- `set_to_fun (hT : dominated_fin_meas_additive μ T C) (f : α → E) : F`: a version of the extension
+  which applies to functions (with value 0 if the function is not integrable).
+
+## Implementation notes
+
+The starting object `T : set α → E →L[ℝ] F` matters only through its restriction on measurable sets
+with finite measure. Its value on other sets is ignored.
+
+The extension step from integrable simple functions to L1 relies on a `second_countable_topology`
+assumption. Without it, we could only extend to `ae_fin_strongly_measurable` functions. (TODO: this
+might be worth doing?)
+
+-/
+
+noncomputable theory
+open_locale classical topological_space big_operators nnreal ennreal measure_theory pointwise
+open set filter topological_space ennreal emetric
+
+local attribute [instance] fact_one_le_one_ennreal
+
+namespace measure_theory
+
+variables {α E F F' G 𝕜 : Type*} {p : ℝ≥0∞}
+  [normed_group E] [measurable_space E] [normed_space ℝ E]
+  [normed_group F] [normed_space ℝ F]
+  [normed_group F'] [normed_space ℝ F']
+  [normed_group G] [measurable_space G]
+  {m : measurable_space α} {μ : measure α}
+
+local infixr ` →ₛ `:25 := simple_func
+
+open finset
+
+section fin_meas_additive
+
+/-- A set function is `fin_meas_additive` if its value on the union of two disjoint measurable
+sets with finite measure is the sum of its values on each set. -/
+def fin_meas_additive {β} [add_monoid β] {m : measurable_space α}
+  (μ : measure α) (T : set α → β) : Prop :=
+∀ s t, measurable_set s → measurable_set t → μ s ≠ ∞ → μ t ≠ ∞ → s ∩ t = ∅ → T (s ∪ t) = T s + T t
+
+lemma map_empty_eq_zero_of_map_union {β} [add_cancel_monoid β] (T : set α → β)
+  (h_add : fin_meas_additive μ T) :
+  T ∅ = 0 :=
+begin
+  have h_empty : μ ∅ ≠ ∞, from (measure_empty.le.trans_lt ennreal.coe_lt_top).ne,
+  specialize h_add ∅ ∅ measurable_set.empty measurable_set.empty h_empty h_empty
+    (set.inter_empty ∅),
+  rw set.union_empty at h_add,
+  nth_rewrite 0 ← add_zero (T ∅) at h_add,
+  exact (add_left_cancel h_add).symm,
+end
+
+lemma map_Union_fin_meas_set_eq_sum {β} [add_comm_monoid β] (T : set α → β) (T_empty : T ∅ = 0)
+  (h_add : fin_meas_additive μ T)
+  {ι} (S : ι → set α) (sι : finset ι) (hS_meas : ∀ i, measurable_set (S i))
+  (hSp : ∀ i ∈ sι, μ (S i) ≠ ∞) (h_disj : ∀ i j ∈ sι, i ≠ j → disjoint (S i) (S j)) :
```

Comment:
There is a standard way in mathlib to express the disjointness assumption, which would read in your case `(h_disj : pairwise_on ↑sι (disjoint on S))`. This is not really important, but sometimes handy as it gives access to a small API and dot notation. I'm not sure it's worth changing here, though, because you already have a working proof.